### PR TITLE
autotest: ensure cached sim time moves forward in frsky test

### PR DESCRIPTION
**Pull request by peterbarker**
https://api.travis-ci.com/v3/job/360309662/log.txt shows this test getting stuck somehow.

The timeout failed to trigger because we're not reading any mavlink data.

It's entirely possible that the ArduPilot process wasn't sending frsky data because it was waiting for the mavlink queue to become less full.

### CODE REVIEW

**Comment by peterbarker** [comment]
@yaapu looks like the frsky test might actually be legitimately busted - it just passes this one depending on how the numbers work out over time.

```
AT-0022.6: validating home status(0x3c004000)
AT-0022.6: GLOBAL_POSITION_INT rel_alt==-74.000000 frsky==1.000000
AT-0022.6: Checking 20487
AT-0022.6: validating params (0x4000ce4)
AT-0022.7: Still wanting (0x5004,0x5007)
FRSkyPassthrough: Sending poll for 0x1b
AT-0022.7: Checking 20484
AT-0022.7: validating home status(0x3c004000)
AT-0022.7: GLOBAL_POSITION_INT rel_alt==-99.000000 frsky==1.000000
AT-0022.7: Checking 20487
AT-0022.7: validating params (0x4000ce4)
AT-0022.8: Still wanting (0x5004,0x5007)
FRSkyPassthrough: status text (dynamic) (0x5000)=1914725217
AT-0022.8: Checking 20484
AT-0022.8: validating home status(0x3c004000)
AT-0022.8: GLOBAL_POSITION_INT rel_alt==-118.000000 frsky==1.000000
AT-0022.8: Checking 20487
AT-0022.8: validating params (0x4000ce4)
AT-0023.0: Still wanting (0x5004,0x5007)
FRSkyPassthrough: Sending poll for 0x1b
AT-0023.0: Checking 20484
AT-0023.0: validating home status(0x3c004000)
AT-0023.0: GLOBAL_POSITION_INT rel_alt==-125.000000 frsky==1.000000
AT-0023.0: Checking 20487
AT-0023.0: validating params (0x4000ce4)
AT-0023.1: Still wanting (0x5004,0x5007)
```

Fixing the timeout as I'm doing here closes the loophole so it's a reliable failure on this PR.

Reproduction with `./Tools/autotest/autotest.py --gdb --debug build.Plane test.Plane.FRSkyPassThrough`

**Comment by yaapu** [comment]
@peterbarker I can't reproduce the failure, I ran it a few times and it always passed.

Peter I'm not sure I understand the issue, what I see is a long slow convergence of relative altitude from GLOBAL_POSITION_INT to the frsky reported one which is locked at 0 or 1 from the very beginning, I also noticed that the frsky scheduler's shaper is time based and packet rates do not take into account the simulation speedup of 8, this explains why params are received so slow.

**Comment by peterbarker** [comment]
@yaapu AFAICS frsky should be reporting the same as `GLOBAL_POSITION_INT` - but I think that's a discussion for another day.

You were running this PR and it didn't fail for you?  That's... odd...

**Comment by peterbarker** [comment]
@yaapu OK, it was a mm vs m thing in the relative-alt test.

Also, needs much, much more time for parameters to come down.  Seems to send the same parameter down *a lot*!

**Comment by yaapu** [comment]
@peterbarker you are right, parameters download code was somehow broken, it sent values for parameters 2,3 and 5 even if not needed, please check if this patch https://github.com/yaapu/ardupilot/tree/pr-fix_frsky_passtrough_params fixes the issue

File: Tools/autotest/common.py [hunk 1]
```diff
@@ -6246,9 +6246,9 @@ def tfp_validate_home_status(self, value):
         )
         if gpi is None:
             raise NotAchievedException("Did not get GLOBAL_POSITION_INT message")
-        gpi_relative_alt = gpi.relative_alt
-        self.progress("GLOBAL_POSITION_INT rel_alt==%f frsky==%f" % (gpi_relative_alt, home_alt_m))
-        if abs(gpi_relative_alt - home_alt_m) < 1:
+        gpi_relative_alt_m = gpi.relative_alt/1000.0
```
Comment (yaapu) [inline Tools/autotest/common.py:6249]: sorry and thanks for catching this one (and mony others)

**Comment by peterbarker** [comment]
@yaapu that branch does seem to be better.  Need to use enum-class and add a comma on the last enum entry, 'though.  Look forward to seeing a PR.

**Pull request merged**